File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/GitFolder.java
package com.oracle.aidataplatform.dp.model;



/**
 * Represents a Git folder object.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=GitFolder.Builder.class)

public final class GitFolder  {
    @Deprecated
    @java.beans.ConstructorProperties({"folderPath", "gitProviderKey", "gitRepositoryUrl", "description", "branchName", "folderMetadata"})
    public GitFolder(String folderPath, String gitProviderKey, String gitRepositoryUrl, String description, String branchName, java.util.Map<String, Object> folderMetadata) {
        super();
        this.folderPath = folderPath;
        this.gitProviderKey = gitProviderKey;
        this.gitRepositoryUrl = gitRepositoryUrl;
        this.description = description;
        this.branchName = branchName;
        this.folderMetadata = folderMetadata;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * The absolute path of the Git folder user wants to create.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("folderPath")
private String folderPath;

        /**
         * The absolute path of the Git folder user wants to create.
         * @param folderPath the value to set
         * @return this builder
         **/
        

public Builder folderPath(String folderPath) {
    this.folderPath = folderPath;
    return this;
}
            /**
     * key corresponding to Git service provider in git provider table.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("gitProviderKey")
private String gitProviderKey;

        /**
         * key corresponding to Git service provider in git provider table.
         * @param gitProviderKey the value to set
         * @return this builder
         **/
        

public Builder gitProviderKey(String gitProviderKey) {
    this.gitProviderKey = gitProviderKey;
    return this;
}
            /**
     * Git repository url used to clone.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("gitRepositoryUrl")
private String gitRepositoryUrl;

        /**
         * Git repository url used to clone.
         * @param gitRepositoryUrl the value to set
         * @return this builder
         **/
        

public Builder gitRepositoryUrl(String gitRepositoryUrl) {
    this.gitRepositoryUrl = gitRepositoryUrl;
    return this;
}
            /**
     * Short description about the git repository.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("description")
private String description;

        /**
         * Short description about the git repository.
         * @param description the value to set
         * @return this builder
         **/
        

public Builder description(String description) {
    this.description = description;
    return this;
}
            /**
     * Git branch name that is cloned.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("branchName")
private String branchName;

        /**
         * Git branch name that is cloned.
         * @param branchName the value to set
         * @return this builder
         **/
        

public Builder branchName(String branchName) {
    this.branchName = branchName;
    return this;
}
            /**
     * The metadata about the folder, like branchName.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("folderMetadata")
private java.util.Map<String, Object> folderMetadata;

        /**
         * The metadata about the folder, like branchName.
         * @param folderMetadata the value to set
         * @return this builder
         **/
        

public Builder folderMetadata(java.util.Map<String, Object> folderMetadata) {
    this.folderMetadata = folderMetadata;
    return this;
}


        public GitFolder build() {
            GitFolder model = new GitFolder(this.folderPath
                , this.gitProviderKey
                , this.gitRepositoryUrl
                , this.description
                , this.branchName
                , this.folderMetadata);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(GitFolder model) {
                this.folderPath(model.getFolderPath());
    this.gitProviderKey(model.getGitProviderKey());
    this.gitRepositoryUrl(model.getGitRepositoryUrl());
    this.description(model.getDescription());
    this.branchName(model.getBranchName());
    this.folderMetadata(model.getFolderMetadata());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * The absolute path of the Git folder user wants to create.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("folderPath")
    private final String folderPath;

        /**
     * The absolute path of the Git folder user wants to create.
     * @return the value
     **/
    
    public String getFolderPath() {
        return folderPath;
    }


        /**
     * key corresponding to Git service provider in git provider table.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("gitProviderKey")
    private final String gitProviderKey;

        /**
     * key corresponding to Git service provider in git provider table.
     * @return the value
     **/
    
    public String getGitProviderKey() {
        return gitProviderKey;
    }


        /**
     * Git repository url used to clone.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("gitRepositoryUrl")
    private final String gitRepositoryUrl;

        /**
     * Git repository url used to clone.
     * @return the value
     **/
    
    public String getGitRepositoryUrl() {
        return gitRepositoryUrl;
    }


        /**
     * Short description about the git repository.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("description")
    private final String description;

        /**
     * Short description about the git repository.
     * @return the value
     **/
    
    public String getDescription() {
        return description;
    }


        /**
     * Git branch name that is cloned.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("branchName")
    private final String branchName;

        /**
     * Git branch name that is cloned.
     * @return the value
     **/
    
    public String getBranchName() {
        return branchName;
    }


        /**
     * The metadata about the folder, like branchName.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("folderMetadata")
    private final java.util.Map<String, Object> folderMetadata;

        /**
     * The metadata about the folder, like branchName.
     * @return the value
     **/
    
    public java.util.Map<String, Object> getFolderMetadata() {
        return folderMetadata;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("GitFolder(");
        sb.append("folderPath=").append(String.valueOf(this.folderPath));
        sb.append(", gitProviderKey=").append(String.valueOf(this.gitProviderKey));
        sb.append(", gitRepositoryUrl=").append(String.valueOf(this.gitRepositoryUrl));
        sb.append(", description=").append(String.valueOf(this.description));
        sb.append(", branchName=").append(String.valueOf(this.branchName));
        sb.append(", folderMetadata=").append(String.valueOf(this.folderMetadata));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof GitFolder)) {
            return false;
        }

        GitFolder other = (GitFolder) o;
        return java.util.Objects.equals(this.folderPath, other.folderPath) &&
            java.util.Objects.equals(this.gitProviderKey, other.gitProviderKey) &&
            java.util.Objects.equals(this.gitRepositoryUrl, other.gitRepositoryUrl) &&
            java.util.Objects.equals(this.description, other.description) &&
            java.util.Objects.equals(this.branchName, other.branchName) &&
            java.util.Objects.equals(this.folderMetadata, other.folderMetadata);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.folderPath == null ? 43 : this.folderPath.hashCode());
        result = (result * PRIME) + (this.gitProviderKey == null ? 43 : this.gitProviderKey.hashCode());
        result = (result * PRIME) + (this.gitRepositoryUrl == null ? 43 : this.gitRepositoryUrl.hashCode());
        result = (result * PRIME) + (this.description == null ? 43 : this.description.hashCode());
        result = (result * PRIME) + (this.branchName == null ? 43 : this.branchName.hashCode());
        result = (result * PRIME) + (this.folderMetadata == null ? 43 : this.folderMetadata.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/git_folder.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class GitFolder(object):
    """
    Represents a Git folder object.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new GitFolder object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param folder_path:
            The value to assign to the folder_path property of this GitFolder.
        :type folder_path: str

        :param git_provider_key:
            The value to assign to the git_provider_key property of this GitFolder.
        :type git_provider_key: str

        :param git_repository_url:
            The value to assign to the git_repository_url property of this GitFolder.
        :type git_repository_url: str

        :param description:
            The value to assign to the description property of this GitFolder.
        :type description: str

        :param branch_name:
            The value to assign to the branch_name property of this GitFolder.
        :type branch_name: str

        :param folder_metadata:
            The value to assign to the folder_metadata property of this GitFolder.
        :type folder_metadata: dict(str, object)

        """
        self.swagger_types = {
            'folder_path': 'str',
            'git_provider_key': 'str',
            'git_repository_url': 'str',
            'description': 'str',
            'branch_name': 'str',
            'folder_metadata': 'dict(str, object)'
        }

        self.attribute_map = {
            'folder_path': 'folderPath',
            'git_provider_key': 'gitProviderKey',
            'git_repository_url': 'gitRepositoryUrl',
            'description': 'description',
            'branch_name': 'branchName',
            'folder_metadata': 'folderMetadata'
        }

        self._folder_path = None
        self._git_provider_key = None
        self._git_repository_url = None
        self._description = None
        self._branch_name = None
        self._folder_metadata = None

    @property
    def folder_path(self):
        """
        **[Required]** Gets the folder_path of this GitFolder.
        The absolute path of the Git folder user wants to create.


        :return: The folder_path of this GitFolder.
        :rtype: str
        """
        return self._folder_path

    @folder_path.setter
    def folder_path(self, folder_path):
        """
        Sets the folder_path of this GitFolder.
        The absolute path of the Git folder user wants to create.


        :param folder_path: The folder_path of this GitFolder.
        :type: str
        """
        self._folder_path = folder_path

    @property
    def git_provider_key(self):
        """
        Gets the git_provider_key of this GitFolder.
        key corresponding to Git service provider in git provider table.


        :return: The git_provider_key of this GitFolder.
        :rtype: str
        """
        return self._git_provider_key

    @git_provider_key.setter
    def git_provider_key(self, git_provider_key):
        """
        Sets the git_provider_key of this GitFolder.
        key corresponding to Git service provider in git provider table.


        :param git_provider_key: The git_provider_key of this GitFolder.
        :type: str
        """
        self._git_provider_key = git_provider_key

    @property
    def git_repository_url(self):
        """
        Gets the git_repository_url of this GitFolder.
        Git repository url used to clone.


        :return: The git_repository_url of this GitFolder.
        :rtype: str
        """
        return self._git_repository_url

    @git_repository_url.setter
    def git_repository_url(self, git_repository_url):
        """
        Sets the git_repository_url of this GitFolder.
        Git repository url used to clone.


        :param git_repository_url: The git_repository_url of this GitFolder.
        :type: str
        """
        self._git_repository_url = git_repository_url

    @property
    def description(self):
        """
        Gets the description of this GitFolder.
        Short description about the git repository.


        :return: The description of this GitFolder.
        :rtype: str
        """
        return self._description

    @description.setter
    def description(self, description):
        """
        Sets the description of this GitFolder.
        Short description about the git repository.


        :param description: The description of this GitFolder.
        :type: str
        """
        self._description = description

    @property
    def branch_name(self):
        """
        Gets the branch_name of this GitFolder.
        Git branch name that is cloned.


        :return: The branch_name of this GitFolder.
        :rtype: str
        """
        return self._branch_name

    @branch_name.setter
    def branch_name(self, branch_name):
        """
        Sets the branch_name of this GitFolder.
        Git branch name that is cloned.


        :param branch_name: The branch_name of this GitFolder.
        :type: str
        """
        self._branch_name = branch_name

    @property
    def folder_metadata(self):
        """
        Gets the folder_metadata of this GitFolder.
        The metadata about the folder, like branchName.


        :return: The folder_metadata of this GitFolder.
        :rtype: dict(str, object)
        """
        return self._folder_metadata

    @folder_metadata.setter
    def folder_metadata(self, folder_metadata):
        """
        Sets the folder_metadata of this GitFolder.
        The metadata about the folder, like branchName.


        :param folder_metadata: The folder_metadata of this GitFolder.
        :type: dict(str, object)
        """
        self._folder_metadata = folder_metadata

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
